core/src/main/java/org/eclipse/hono/util/HonoProtonHelper.java

```diff
@@ -60,7 +65,11 @@ private HonoProtonHelper() {
         Objects.requireNonNull(handler);
 
         final Handler<AsyncResult<T>> wrappedHandler = remoteDetach -> {
-            handler.handle(remoteDetach);
+            try {
+                handler.handle(remoteDetach);
+            } catch (final Exception ex) {
+                LOG.warn("error running detachHandler", ex);
+            }
             link.free();
```

Comment:
didn't one of the stack traces you showed me that a runtime exception had been thrown while executing the *free()* method?

Reply:
yes, but now these exceptions in *free()* shouldn't happen anymore.